Core/Core/Grades/Model/GradeListInteractor.swift

```diff
@@ -127,10 +126,8 @@ public final class GradeListInteractorLive: GradeListInteractor {
             let enrollments = params.2
             let courseEnrollment = course.enrollmentForGrades(userId: userID)
             let isGradingPeriodHidden = courseEnrollment?.multipleGradingPeriodsEnabled == false
-
             if !isInitialGradingPeriodSet {
                 isInitialGradingPeriodSet = true
-                updateGradingPeriod(id: courseEnrollment?.currentGradingPeriodID)
```

Comment:
Removed this function because called of from viewModel so can get all the  Grading Period,

Reply:
I'm not sure about removing this. By removing it, the interactor itself won't be able to correctly get the grades because `updateGradingPeriod` is only called from the view model, right?

Reply:
@szabinst yes, you are right but i removed this from interactor and make it loaded only one time from viewModel, because calling it from interactor not giving me the correct grade periods at first time, And i have checked it is working fine like the pervious implementation.

Reply:
And does the interactor works properly now if I let's say, use it at with an other screen, view model?

Reply:
Fixed a pear our call